airbyte-integrations/connectors/source-hubspot/source_hubspot/api.py

```diff
@@ -589,31 +593,39 @@ def read(self, getter: Callable, params: Mapping[str, Any] = None) -> Iterator:
         payload = (
             {
                 "filters": [{"value": int(self._state.timestamp() * 1000), "propertyName": self.last_modified_field, "operator": "GTE"}],
+                "sorts": [{"propertyName": self.last_modified_field, "direction": "ASCENDING"}],
                 "properties": properties_list,
-                "limit": 100,
+                "limit": self.limit,
             }
             if self.state
             else {}
         )
 
-        while True:
-            stream_records = {}
-            if self.state:
+        after = 0
+        if self.state:
```

Comment:
the logic for stateless and stateful paths start to diverge quite a bit and doesn't feel very pythonic. Open to suggestion how to improve this.